common/metrics/metricstest/metricstest.go

```diff
@@ -193,10 +244,52 @@ func (s Snapshot) MustGauge(name string, tags ...metrics.Tag) float64 {
 	return v
 }
 
+func (s Snapshot) Histogram(name string, tags ...metrics.Tag) ([]HistogramBucket, error) {
+	labelValues := map[string]string{}
+	for _, tag := range tags {
+		labelValues[tag.Key()] = tag.Value()
+	}
+
+	sample, ok := s.histogramSamples[name]
+	if !ok {
+		return nil, fmt.Errorf("metric %s not found", name)
+	}
+	if sample.metricType != dto.MetricType_HISTOGRAM {
+		return nil, fmt.Errorf("metric %s not a %s type", name, dto.MetricType_HISTOGRAM.String())
+	}
+	if !maps.Equal(sample.labelValues, labelValues) {
+		return nil, fmt.Errorf("metric %s label mismatch, has %v, asked for %v", name, sample.labelValues, labelValues)
```

Comment:
I think we should actually just ignore this sample instead of returning an error because that's how PromQL actually works. We should be operating on the samples whose labels match